test/tests/storage/daemon_agent/main.go

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2020 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"time"
+
+	"github.com/gitpod-io/gitpod/content-service/pkg/storage"
+	"github.com/gitpod-io/gitpod/test/pkg/integration"
+	"github.com/gitpod-io/gitpod/test/tests/storage/daemon_agent/api"
+)
+
+func main() {
+	integration.ServeAgent(new(DaemonAgent))
+}
+
+type daemonConfig struct {
+	Daemon struct {
+		Content struct {
+			Storage storage.Config `json:"storage"`
+		} `json:"content"`
+	} `json:"daemon"`
+}
+
+// DaemonAgent provides ingteration test services from within ws-daemon
+type DaemonAgent struct {
+}
+
+// CreateBucket reads the daemon's config, and creates a bucket
+func (*DaemonAgent) CreateBucket(args *api.CreateBucketRequest, resp *api.CreateBucketResponse) error {
```

Comment:
Indeed, that's the value of this test.

The trade-off/decision is exactly the one you outline: either embed the testing functionality into the components and make them available their gRPC interface, or we keep this logic out of the components and use agents instead. In this particular case I'm really really on the fence, because arguably trying to create a bucket is a self-test that ws-daemon could certainly run at start-up. Maybe this use of agents just isn't a good one.

My reasoning was that we have agents already, might as well use them rather than extend other components. I reckon there are other cases which are not as close a call, e.g. meddling with the on-disk state of ws-daemon. Certainly this could be done in a `TestService`, but it would be super intrusive to the component and spread the test out across several components (test and ws-daemon).

Reply:
> In this particular case I'm really really on the fenc

> I reckon there are other cases which are not as close a call,

Thank you for elaborating; completely agree!